src/bitboard.h

```diff
@@ -82,20 +82,17 @@ struct Magic {
   Bitboard  mask;
   Bitboard  magic;
   Bitboard* attacks;
-  unsigned  shift;
 
   // Compute the attack's index using the 'magic bitboards' approach
+  template<PieceType Pt>
   unsigned index(Bitboard occupied) const {
 
     if (HasPext)
         return unsigned(pext(occupied, mask));
 
-    if (Is64Bit)
-        return unsigned(((occupied & mask) * magic) >> shift);
+    unsigned shift = 64 - (Pt == ROOK ? 12 : 9);
 
-    unsigned lo = unsigned(occupied) & unsigned(mask);
-    unsigned hi = unsigned(occupied >> 32) & unsigned(mask >> 32);
-    return (lo * unsigned(magic) ^ hi * unsigned(magic >> 32)) >> shift;
+    return unsigned(((occupied & mask) * magic) >> shift);
```

Comment:
Won't this be a speed regression for 32 bit ?
Actually, do we still care about 32 bit performance ? (I don't, but I remember that @mcostalba was very adamant about it).

Reply:
It seems that on the machines I tested the gains of this patch make up for the removal of the specialized multiplication.